index.js

```diff
@@ -266,8 +268,20 @@ class K8sVMExecutor extends Executor {
             LOW: this.lowCpu,
             MICRO: this.microCpu
         };
-        // set pipeline cache volume readonly for PRs
-        const volumeReadOnly = jobName.slice(0, 3) === 'PR-';
+
+        // PRs - set pipeline, job cache volume readonly and job cache dir to parent job cache dir
+        const regex = /^PR-([0-9]+)(?::[\w-]+)?$/gi;
+        const matched = regex.exec(jobName);
+        let volumeReadOnly = false;
+
+        if (matched && matched.length === 2) {
+            const decodedToken = jwt.decode(token, { complete: true });
+
+            volumeReadOnly = true;
+            jobId = hoek.reach(decodedToken.payload,
```

Comment:
what's the implication of overriding jobId itself ? Won't this cause problems for further processing of the job? For example now `launcher` might set parentJobId as jobId for PR builds.

Reply:
No, it will not, this is used only for cache dir. launcher or store-cli doesnt depend on this.

Reply:
cool, my only worry is if in future if someone misuses this variable and see unexpected results.

Reply:
agree, verified we don't pass jobId to store-cli and launcher, only the dir is passed.